loopchain/blockchain/blockchain.py

```diff
@@ -636,6 +654,9 @@ def confirm_prev_block(self, current_block: Block):
             except KeyError:
                 if self.last_block.header.hash == current_block.header.prev_hash:
                     logging.warning(f"Already added block hash({current_block.header.prev_hash.hex()})")
+                    if current_block.header.is_complain:
+                        util.logger.debug("reset last_unconfirmed_block by complain block")
+                        self.last_unconfirmed_block = current_block
```

Comment:
The method has responsibility for only `confirm_prev_block`.
This exception handling is located out of the method.